### PR TITLE
Avoid overflow in addition of bspline offset for integer-type converters

### LINKED ISSUE #178: Unsigned Integer Overflow
While converting an uint16 image to APR format and converting back to tiff, it seems the some zero background pixels are arbitrarily changed to 65535. My images are uint16, with some pixels exactly 0.

Here are the commands I am using,
```
LibAPR/build/examples/Example_get_apr -i input.tif -o test -lambda 0
```
and
```
LibAPR/build/examples/Example_reconstruct_image -i test.apr -o test
```
When pixels are non-zero, the difference between my input.tif and output test_pc.tif are small, except for zero pixels being randomly changed to 65535.
I am happy to provide examples if this is not reproducible.

### PR DESCRIPTION
**Pull request by joeljonsson**
Resolves #178 

The bspline offset is now computed based on the range of values in the input image, and only if smoothing is enabled (`lambda>0`). 

Small revision of examples:
- `Example_get_apr*` now use converters of type `float`.
- Added `-auto_parameters` option to `Example_get_apr_by_block`
- Addition of gaussian noise to the piecewise constant reconstruction in `Example_reconstruct_image` is now controlled via the option `-noise <sigma>` where the standard deviation can be set (default 0 = no additional noise)

### CODE REVIEW

**Comment by cheesema** [comment]
So the current failing test seem to be down to:

https://github.com/cheesema/AdaptiveParticleRepresentation/blob/40e807b87e8d57f1cc4ae14a9c3f6a38286c61ca/src/algorithm/LocalParticleCellSet.hpp#L136

In the sphere2D case; the local intensity scale is exactly 0. This results in nan's, it seem like all other systems this results in 0's for the downstream level function. However, in macosx, it seems this now behaves differently. 

Tests passing can be recovered by adding 1 to the 0 entries in the LIS. 

I am reluctant to add another operation in main, to all paths to handle this case without thought. 

@joeljonsson what do you think?

**Comment by cheesema** [comment]
So it seemed the thesholds on this example were set to 0's. Which should never really be the case, but did not seem to matter up to this point.

**Comment by joeljonsson** [comment]
It seems that all test files are computed with `sigma_th=0` and `sigma_th_max=0`. Clearly having tests rely on undefined behavior giving a certain result is a bad idea. This might be the case in other tests as well, so it is probably best to replace all of the APR test files (and corresponding `output_steps` images), e.g. with ones computed using autoparameters. 

As discussed before we should also add a warning, e.g. [here](https://github.com/AdaptiveParticles/LibAPR/blob/40e807b87e8d57f1cc4ae14a9c3f6a38286c61ca/src/algorithm/APRConverter.hpp#L249), when both `sigma_th` and `sigma_th_max` are `0` that this may lead to undefined behavior.

**Comment by cheesema** [comment]
Added  the  warning and a  test. To be honest; changing all the test files seems more dangerous, let along more work; then this current fix for now.

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,14 +1,18 @@
...
-cmake_minimum_required(VERSION 3.10)
+cmake_minimum_required(VERSION 3.14)
 project(APR DESCRIPTION "Adaptive Particle Representation library")
 
 message(STATUS "CMAKE VERSION ${CMAKE_VERSION}")
 
 set(CMAKE_CXX_STANDARD 14)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
+if(POLICY CMP0135)
```
Comment (cheesema) [inline CMakeLists.txt:12]: Why was this needed to  be added?

**Review by cheesema — APPROVED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (joeljonsson) [reply CMakeLists.txt:12]: Honestly I'm not sure, but I think it has to do with the way gtest is now downloaded
